### ISSUE TITLE
trait supertrait is not rendered

### ISSUE THREAD

**Issue by Emilgardis** [cargo-public-api/cargo-public-api#151]
the following code

```rust
pub trait T: private::Sealed + Send {}

impl T for () {}

mod private {
    pub trait Sealed {}
    impl Sealed for () {}
}
```

generates

```
pub mod krate
pub trait krate::T
```

the supertrait `private::Sealed + Send` should be rendered here.

one side note also, the implementor `T for ()` should be rendered, is there an issue for that available?

**Comment by Emilgardis** [comment]
Ideally, we'd also render `impl Sealed for () {}` as well

**Comment by Enselic** [comment]
👍

I have started working on also rendering impls, but it is not ready yet: https://github.com/Enselic/cargo-public-api/pull/152

**Comment by Emilgardis** [comment]
> Ideally, we'd also render `impl Sealed for () {}` as well

This is currently not implemented, should a new issue be created for that maybe?

**Comment by Enselic** [comment]
I'm thinking that's part of #36, even though the title does not perfectly match.

I am currently prototyping what the code should look like to render all `impl`s in a public API. Right now it looks like each `PublicItem` will (pseudocode) get a field `impls: Vec<PublicItem>` and `children: Vec<PublicItem>` so that for a `pub struct Foo` the `cargo public-api` output would be

```rust
pub struct Foo
impl Display for Foo {      // <-- in `impls` for `PublicItem` for `Foo`
    fn fmt(etc)             // <-- in `children` for `PublicItem` for the `impl`
}
```

and in the future we could use `children` to render struct fields as part of structs:

```rust
pub struct Foo {
    pub a_field: bool,
}
```

The above feels promising to me even though this is only at early stages of experimentation.

Your case with `impl Sealed for () {}` is extra tricky since I don't think the rustdoc JSON contains an item for `()` with its list of `impl`s. But I think `Sealed` contains a list of `implementors`. But maybe we need to create a follow-up ticket later.

The above work is intended to also make progress on or fix #37 and #39

**Comment by Enselic** [comment]
After https://github.com/Enselic/cargo-public-api/pull/195 lands we will start to render impls for (). Using the code in that PR, here is the public API listing for

```rust
pub trait T: private::Sealed + Send {}

impl T for () {}

mod private {
    pub trait Sealed {}
    impl Sealed for () {}
}
```

```
$ cargo run -- --manifest-path ~/src/lib/Cargo.toml --unsorted 
pub mod api_test
pub trait api_test::T: private::Sealed + Send
impl T for ()
```
